java/src/com/google/idea/blaze/java/fastbuild/FastBuildChangedFilesService.java

```diff
@@ -231,25 +235,39 @@ public void after(List<? extends VFileEvent> events) {
                     return null;
                   }
 
-                  ImmutableSet<File> changedFiles =
+                  ImmutableSet<String> changedFilePaths =
                       events.stream()
                           // We don't want to compile deleted files
                           .filter(event -> !(event instanceof VFileDeleteEvent))
                           .map(VFileEvent::getPath)
+                          .collect(toImmutableSet());
+
+                  ImmutableSet<File> changedCompilableFiles =
+                      changedFilePaths.stream()
                           .filter(f -> f.endsWith(".java") || f.endsWith(".scala"))
                           .map(File::new)
                           .collect(toImmutableSet());
 
-                  if (changedFiles.isEmpty()) {
-                    return null;
-                  }
+                  ImmutableSet<File> changedProtoFiles =
+                      changedFilePaths.stream()
+                          .filter(f -> f.endsWith(".proto"))
+                          .map(File::new)
+                          .collect(toImmutableSet());
+
                   // TODO(b/145386688): Access should be guarded by enclosing instance
                   // 'com.google.idea.blaze.java.fastbuild.FastBuildChangedFilesService' of 'data',
                   // which is not accessible in this scope
-                  labelData.values().forEach(data -> data.updateChangedSources(changedFiles));
+
+                  if (!changedCompilableFiles.isEmpty() || !changedProtoFiles.isEmpty()) {
```

Comment:
is there a simpler form for this if? I needed to read it a few times (maybe it's just me though 🤦 )

Reply:
!(changedCompilableFiles.isEmpty() && changedProtoFiles.isEmpty()) is always an alternative :D

Reply:
🤷 leave it as is I guess. 
I thought maybe a private method would help for readability but maybe it won't...